tests/ui/pattern/rfc-3627-match-ergonomics-2024/migration_lint.stderr

```diff
@@ -0,0 +1,188 @@
+error: patterns are not allowed to reset the default binding mode in edition 2024
+  --> $DIR/migration_lint.rs:25:9
+   |
+LL |     let Foo(mut x) = &Foo(0);
+   |         -^^^^^^^^^
+   |         |
+   |         help: desugar the match ergonomics: `&`
```

Comment:
I think this message may be a little bit too obtuse for the average reader.

In particular, my guess is that the average person leveraging match ergnonomics may not even *know* that they are doing so, and so they will not necessarily understand what this help message is telling them to do.

I recommend rephrasing it with a little more direction. Perhaps: "add a `&` to the pattern you have written to correspond to the type of the expression being matched" ?

Reply:
oh yeah excellent point